pkg/dataobj/sections/pointers/metrics.go

```diff
@@ -261,10 +261,11 @@ func (m *Metrics) Unregister(reg prometheus.Registerer) {
 // Observe observes section statistics for a given section.
 func (m *Metrics) Observe(ctx context.Context, section *Section) error {
 	dec := newDecoder(section.reader)
-	columns, err := dec.Columns(ctx)
+	metadata, err := dec.Metadata(ctx)
 	if err != nil {
 		return err
 	}
+	columns := metadata.GetColumns()
```

Comment:
nit: Noticed that we name the variable from the return value of `metadata.GetColumns()` differently: `cols`, `columns`, and `columnDescs`.
Maybe we should unify them?